src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -194,6 +214,92 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 
 
 /*
+ * ReplaceExternParamsOnOriginalQuery replaces the external parameters that appears
+ * on the query with the corresponding entries on the boundParams.
+ *
+ * Note that this function is inspired by eval_const_expr() on Postgres. We failed to
+ * use that function given that it requires access to PlannerInfo.
+ */
+static Node *
+ReplaceExternParamsOnOriginalQuery(Node *inputNode, ParamListInfo boundParams)
```

Comment:
I think we probably don't want to use `ExecuteMasterEvaluableFunctions ()` given that we don't want to evaluate other stuff except for the PARAMs on the `originalQuery`.

I've updated the function comment (which I was doing even before your comment 😄 )